tests/test_flags.py

```diff
@@ -0,0 +1,332 @@
+        assert not ins != other
+
+        ins.two = False
+        assert ins is not other
+        assert not ins == other
+        assert ins != other
+
+    def test__and__(self) -> None:
+        ins = TestFlags(one=True, two=True)
+        other = TestFlags(one=True, two=True)
+
+        third = ins & other
+        assert third is not ins
+        assert third.value == 0b011
+        assert third == ins == other
+
+        ins.one = False
+        third = ins & other
+        assert third is not ins
+        assert third.value == 0b010
+
+    def test__iand__(self) -> None:
+        ins = TestFlags(one=True, two=True)
+        other = TestFlags(one=True, two=True)
+
+        third = ins
+        ins &= other
+        assert third is ins
+        assert ins.value == 0b011
+
+        other.two = False
+        other.four = True
+        ins &= other
+        assert third is ins
+        assert ins.value == 0b001
+
+    def test__or__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        third = ins | other
+        assert third is not ins
+        assert ins.value == 0b001
+        assert other.value == 0b010
+        assert third.value == 0b011
+
+        ins.one = False
+        third = ins | other
+        assert third.value == 0b010
+        assert third is not ins
+
+        ins.value = 0b10
+        other.value = ins.value
+        third = ins | other
+        assert third is not ins
+        assert third.value == 0b10
+
+    def test__ior__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        third = ins
+        ins |= other
+        assert ins is third
+        assert ins.value == 0b011
+        assert other.value == 0b010
+
+        other.four = True
+        ins |= other
+        assert ins.value == 0b111
+
+    def test__xor__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        third = ins ^ other
+        assert third.value == 0b011
+        assert third is not ins
+
+        other.one = True
+        third = ins ^ other
+        assert third.value == 0b010
+
+    def test__ixor__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        third = ins
+        ins ^= other
+        assert ins is third
+        assert ins.value == 0b011
+
+        ins.two = False
+        other.one = True
+        ins ^= other
+        assert ins.value == 0b010
+
+    def test__le__(self) -> None:
+        ins = TestFlags(one=True, two=False)
+        other = TestFlags(one=False, two=True)
+
+        assert not ins <= other
+        other.one = True
+        assert ins <= other
+
+        with pytest.raises(
+            TypeError, match="'<=' not supported between instances of 'TestFlags' and 'int'"
+        ):
+            ins <= 4  # type: ignore  # noqa: B015
```

Comment:
```suggestion
            _ = ins <= 4  # type: ignore
```
I know I suggested the other way before, but it's better to keep it consistent with the other ones (whichever way the comparison result is handled)

Reply:
https://github.com/DisnakeDev/disnake/pull/616/commits/312d762e43508ef8dd002bc6cb0c1d79196083aa